tools/shared/src/main/scala/org/scalajs/core/tools/linker/backend/emitter/FunctionEmitter.scala

```diff
@@ -625,6 +625,17 @@ private[emitter] class FunctionEmitter(jsGen: JSGen) {
             }, newLabel)
           }
 
+        case ForIn(obj, keyVar, body) =>
+          unnest(obj) { (newObj, env0) =>
+            implicit val env = env0
+
+            val lhs = genEmptyImmutableLet(transformLocalVarIdent(keyVar))
+            js.ForIn(lhs, transformExprNoChar(newObj), {
```

Comment:
Having an immutable let as lhs here is IMO a bit of an abuse (just like we used to have for LHSes in the desugarer), but I can't come up with anything better.

Reply:
Same here. 😞